radeon-profile/gpu.cpp

```diff
@@ -265,12 +265,24 @@ void gpu::setForcePowerLevel(const QString &newForcePowerLevel) {
 }
 
 void gpu::setPwmValue(unsigned int value) {
+    // If the PC is sent to sleep (or hibernate) it can happen that PWM is
+    // disabled (by the OS..?) and that we have to re-enable it, if needed,
+    // after returning from sleep
+    static QFile pwmEnableFile(getDriverFiles().hwmonAttributes.pwm1_enable);
+    if (Q_UNLIKELY(pwmEnabled && pwmEnableFile.open(QFile::ReadOnly)
+                   && !pwmEnableFile.read(1).contains(pwm_manual)))
+    {
+        setPwmManualControl(true);
+    }
+    pwmEnableFile.close();
+
     value = getGpuConstParams().pwmMaxSpeed * value / 100;
     driverHandler->setNewValue(getDriverFiles().hwmonAttributes.pwm1, QString::number(value));
 }
 
 void gpu::setPwmManualControl(bool manual) {
     driverHandler->setNewValue(getDriverFiles().hwmonAttributes.pwm1_enable, QString(manual ? pwm_manual : pwm_auto));
+    pwmEnabled = manual;
```

Comment:
Is this variable needed? I think that after resume value will be the same as when system was suspended so it won't check the file. I think just checking the file will be ok.

Reply:
Yep, no problem. I'll get rid of this variable then